### ISSUE TITLE
Cannot get figures and tables with airpy.summarize

### ISSUE THREAD

**Issue by chdegrave** [jlandercy/airpy#29]
I try to use airpy.summarize with this command : 

```
python -m airpy.summarize --file ntu_2012_2016.hdf5 --node /timeserie/DD_2012_2016 --figures timeserie --granulityid 3 --mode dataset --clusters sitekey measurandkey
```

And I get this message : 

```
2017-06-19 15:56:13,270 chdegrave@L7_FR_16459[10.0.1.31]        MainThread      airpy\summarize.py  main        INFO    Parsing cluster:        sitekey
Traceback (most recent call last):
  File "C:\Program Files\Anaconda3\lib\runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Program Files\Anaconda3\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\summarize.py", line 149, in <module>
    main(sys.argv)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\clargs.py", line 30, in inner
    func = f(*args, **kwargs)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\summarize.py", line 124, in main
    axe = timeserie.plot(ds)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\report\timeserie.py", line 50, in plot
    ylim = [limits.min()[0], limits.max()[1]]
TypeError: 'float' object is not subscriptable
```

It comes from the channels without ylim ?

**Comment by jlandercy** [comment]
Bug reproduced with the latest version.

It happens because there is no `axismilits` defined in database for your channel selection. Then `min()` aggregates cast `None`  into `Nan` which is a `float` not a `list`, leading to `TypeError`, `float` cannot be indexed with `[]` operator. 

```
networkkey  measurandkey  sitekey  unitskey  channelid
HMT         As            01AND3   ng/m3     5154         None
            Cd            01AND3   ng/m3     5178         None
            Cr            01AND3   ng/m3     5166         None
            Cu            01AND3   ng/m3     5119         None
            Mn            01AND3   ng/m3     5222         None
            Ni            01AND3   ng/m3     5136         None
            Pb            01AND3   ng/m3     5233         None
            SO4(aq)       01AND3   ug/m3     5244         None
            Sb            01AND3   ng/m3     5211         None
            Tl            01AND3   ng/m3     5190         None
Name: axislimits, dtype: object
Series([], Name: axislimits, dtype: object)
nan
Traceback (most recent call last):
  File "C:\Program Files\Anaconda3\lib\runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Program Files\Anaconda3\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\summarize.py", line 157, in <module>
    main(sys.argv)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\clargs.py", line 30, in inner
    func = f(*args, **kwargs)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\summarize.py", line 132, in main
    axe = timeserie.plot(ds)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\report\timeserie.py", line 53, in plot
    ylim = [limits.min()[0], limits.max()[1]]
TypeError: 'float' object is not subscriptable
```

Nota: I have seen that your version is outdated (please when submitting bugs, ensure you have the latest version).

Starting hotfix for your issue.

**Comment by jlandercy** [comment]
Partial fix (0d01b849a53e20e07a7daa01fcf16440c2cb4aba), now timeserie works, but scatter crashes:
```

Traceback (most recent call last):
  File "C:\Program Files\Anaconda3\lib\runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Program Files\Anaconda3\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\summarize.py", line 157, in <module>
    main(sys.argv)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\clargs.py", line 30, in inner
    func = f(*args, **kwargs)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\summarize.py", line 141, in main
    axe = scatter.plot(ds)
  File "C:\Program Files\Anaconda3\lib\site-packages\airpy-1.525-py3.5.egg\airpy\report\scatter.py", line 66, in plot
    xlim = meta.limits.loc[k1]
  File "C:\Program Files\Anaconda3\lib\site-packages\pandas\core\indexing.py", line 1294, in __getitem__
    return self._getitem_tuple(key)
  File "C:\Program Files\Anaconda3\lib\site-packages\pandas\core\indexing.py", line 789, in _getitem_tuple
    self._has_valid_tuple(tup)
  File "C:\Program Files\Anaconda3\lib\site-packages\pandas\core\indexing.py", line 141, in _has_valid_tuple
    raise IndexingError('Too many indexers')
pandas.core.indexing.IndexingError: Too many indexers
```

**Comment by jlandercy** [comment]
Partial fix (49bce5cb5174da0e15035e5e2a5536b76bf53fe1), but histogram is buggy and silenced. Some scatters does not render.

**Comment by jlandercy** [comment]
Fixed, timeserie and scatter work again:
- [x] Dataset `subsample` and `purge` methods added (605d57083ac2cb37252df200faaaa26d213dfc7d, 20ae62375bb579713b4ed88012d250760d405c56);
- [x] Dataset metadata `extremelimits` method added (747b175de4a124cf8205fbeab772aebcecacb576);
- [x] Removed old code in scatter that was causing another error with single column dataset (fba535e40bf6c8ef877cb5a3fadb280a42884a4f).

**Comment by jlandercy** [comment]
Bug #29 corrected, see merge #31
